src/coreComponents/mesh/utilities/ComputationalGeometry.hpp

```diff
@@ -446,47 +450,15 @@ void getBoundingBox( localIndex const elemIndex,
  */
 GEOSX_HOST_DEVICE
 inline
-real64 hexVolume( real64 const X[][3] )
+real64 hexVolume( real64 const (&X)[8][3] )
 {
-  real64 X7_X1[ 3 ] = LVARRAY_TENSOROPS_INIT_LOCAL_3( X[7] );
-  LvArray::tensorOps::subtract< 3 >( X7_X1, X[1] );
-
-  real64 X6_X0[ 3 ] = LVARRAY_TENSOROPS_INIT_LOCAL_3( X[6] );
-  LvArray::tensorOps::subtract< 3 >( X6_X0, X[0] );
-
-  real64 X7_X2[ 3 ] = LVARRAY_TENSOROPS_INIT_LOCAL_3( X[7] );
-  LvArray::tensorOps::subtract< 3 >( X7_X2, X[2] );
-
-  real64 X3_X0[ 3 ] = LVARRAY_TENSOROPS_INIT_LOCAL_3( X[3] );
-  LvArray::tensorOps::subtract< 3 >( X3_X0, X[0] );
-
-  real64 X5_X0[ 3 ] = LVARRAY_TENSOROPS_INIT_LOCAL_3( X[5] );
-  LvArray::tensorOps::subtract< 3 >( X5_X0, X[0] );
-
-  real64 X7_X4[ 3 ] = LVARRAY_TENSOROPS_INIT_LOCAL_3( X[7] );
-  LvArray::tensorOps::subtract< 3 >( X7_X4, X[4] );
-
-  real64 X7_X1plusX6_X0[ 3 ] = LVARRAY_TENSOROPS_INIT_LOCAL_3( X7_X1 );
-  LvArray::tensorOps::add< 3 >( X7_X1plusX6_X0, X6_X0 );
-
-  real64 X7_X2plusX5_X0[ 3 ] = LVARRAY_TENSOROPS_INIT_LOCAL_3( X7_X2 );
-  LvArray::tensorOps::add< 3 >( X7_X2plusX5_X0, X5_X0 );
-
-  real64 X7_X4plusX3_X0[ 3 ] = LVARRAY_TENSOROPS_INIT_LOCAL_3( X7_X4 );
-  LvArray::tensorOps::add< 3 >( X7_X4plusX3_X0, X3_X0 );
-
-  real64 X7_X2crossX3_X0[3];
-  LvArray::tensorOps::crossProduct( X7_X2crossX3_X0, X7_X2, X3_X0 );
-
-  real64 X7_X2plusX5_X0crossX7_X4[3];
-  LvArray::tensorOps::crossProduct( X7_X2plusX5_X0crossX7_X4, X7_X2plusX5_X0, X7_X4 );
-
-  real64 X5_X0crossX7_X4plusX3_X0[3];
-  LvArray::tensorOps::crossProduct( X5_X0crossX7_X4plusX3_X0, X5_X0, X7_X4plusX3_X0 );
-
-  return 1.0/12.0 * ( LvArray::tensorOps::AiBi< 3 >( X7_X1plusX6_X0, X7_X2crossX3_X0 ) +
-                      LvArray::tensorOps::AiBi< 3 >( X6_X0, X7_X2plusX5_X0crossX7_X4 ) +
-                      LvArray::tensorOps::AiBi< 3 >( X7_X1, X5_X0crossX7_X4plusX3_X0 ) );
```

Comment:
@castelletto1 You will not find a more accurate method of calculating the volume of a hex than this one. I have a Mathematica script somewhere that proves this.

Reply:
I don't question that the current implementation based on the [tetrakis hexahedron](https://en.wikipedia.org/wiki/Tetrakis_hexahedron) is extremely efficient (see report [Efficient Computation of Volume of
Hexahedral Cells](https://www.osti.gov/servlets/purl/632793) by J. Grandy). However, the fundamental assumption is that hexahedra are treated using triangular boundaries. Conversely, my goal is to compute the volume of hexahedra, and also of wedges and pyramids, accounting for possible non planar (bilinear) faces. This is consistent with what is done in the finite element solvers. If a finite element based solver is coupled with a finite volume solver, and different kernels are used to assemble cell-wise contribution, it is important to ensure that cell volumes are always the same.